### ISSUE TITLE
[Bug] vits best_model.pth probably not saving the best model

### ISSUE THREAD

**Issue by mobassir94** [coqui-ai/TTS#2104]
### Describe the bug

i was training vits from scratch for bangla. the best_model.pth was saved after 11648 global steps and after that the best_model.pth got never updated again. during inference i've realized that best_model.pth is not good enough but on the other hand checkpoint_313000.pth is giving me far better synthesized audio than the best_model.pth and now i am confused when to stop training. i am training vits on A6000 since last 6 days and best_model.pth was updated 5 days ago and after that it never gets updated.

this is the config i used during training vits

```
{
    "output_path": "/home/ansary/Shabab/",
    "logger_uri": null,
    "run_name": "vits_20_october",
    "project_name": null,
    "run_description": "\ud83d\udc38Coqui trainer run.",
    "print_step": 50,
    "plot_step": 100,
    "model_param_stats": false,
    "wandb_entity": null,
    "dashboard_logger": "tensorboard",
    "log_model_step": null,
    "save_step": 1000,
    "save_n_checkpoints": 5,
    "save_checkpoints": true,
    "save_all_best": false,
    "save_best_after": 10000,
    "target_loss": null,
    "print_eval": false,
    "test_delay_epochs": -1,
    "run_eval": true,
    "run_eval_steps": null,
    "distributed_backend": "nccl",
    "distributed_url": "tcp://localhost:54321",
    "mixed_precision": true,
    "epochs": 10000,
    "batch_size": 96,
    "eval_batch_size": 40,
    "grad_clip": [
        1000,
        1000
    ],
    "scheduler_after_epoch": true,
    "lr": 0.001,
    "optimizer": "AdamW",
    "optimizer_params": {
        "betas": [
            0.8,
            0.99
        ],
        "eps": 1e-09,
        "weight_decay": 0.01
    },
    "lr_scheduler": "",
    "lr_scheduler_params": null,
    "use_grad_scaler": false,
    "cudnn_enable": true,
    "cudnn_deterministic": false,
    "cudnn_benchmark": true,
    "training_seed": 54321,
    "model": "vits",
    "num_loader_workers": 4,
    "num_eval_loader_workers": 4,
    "use_noise_augment": false,
    "audio": {
        "fft_size": 1024,
        "sample_rate": 22050,
        "win_length": 1024,
        "hop_length": 256,
        "num_mels": 80,
        "mel_fmin": 0,
        "mel_fmax": null
    },
    "use_phonemes": false,
    "phonemizer": null,
    "phoneme_language": null,
    "compute_input_seq_cache": true,
    "text_cleaner": null,
    "enable_eos_bos_chars": false,
    "test_sentences_file": "",
    "phoneme_cache_path": null,
    "characters": {
        "characters_class": "TTS.tts.utils.text.characters.Graphemes",
        "vocab_dict": null,
        "pad": "<PAD>",
        "eos": "\u0964",
        "bos": "<BOS>",
        "blank": "<BLNK>",
        "characters": "\u09a4\u099f\u09eb\u09ad\u09bf\u0990\u098b\u0996\u098a\u09dc\u0987\u099c\u09ae\u098f\u09c7\u0998\u0999\u09b8\u09c0\u09dd\u09b9\u099e\u2018\u0988\u0995\u09a3\u09ec\u0981\u09d7\u09b6\u09a2\u09a0\u200c\u09e7\u09cd\u09e8\u09ee\u09a6\u09c3\u0994\u0997\u0993\u2014\u099b\u0989\u0982\u09ac\u09c8\u099d\u09be\u09af\u09ab\u200d\u099a\u09b0\u09b7\u0985\u09cc\u09ce\u09a5\u09a1\u09bc\u09ea\u09a7\u09e6\u09c1\u09c2\u09e9\u0986\u0983\u09aa\u09df\u2019\u09a8\u09b2\u09cb",
        "punctuations": "-!,|.? ",
        "phonemes": null,
        "is_unique": true,
        "is_sorted": true
    },
    "add_blank": true,
    "batch_group_size": 0,
    "loss_masking": null,
    "min_audio_len": 1,
    "max_audio_len": Infinity,
    "min_text_len": 1,
    "max_text_len": Infinity,
    "compute_f0": false,
    "compute_linear_spec": true,
    "precompute_num_workers": 0,
    "start_by_longest": false,
    "datasets": [
        {
            "name": "",
            "path": "/home/ansary/Shabab/bn_tts/male/mono/",
            "meta_file_train": "/home/ansary/Shabab/bn_tts/male/mono/metadata_male.txt",
            "ignored_speakers": null,
            "language": "",
            "meta_file_val": "",
            "meta_file_attn_mask": ""
        }
    ],
    "test_sentences": [
        "\u09aa\u09bf\u09aa\u09b2\u09b8 \u0987\u09a8\u09cd\u09b8\u09cd\u09af\u09c1\u09b0\u09c7\u09a8\u09cd\u09b8 \u0985\u09ac \u099a\u09be\u09df\u09a8\u09be \u099b\u09c7\u09b7\u099f\u09cd\u099f\u09bf \u09ac\u099b\u09b0 \u0986\u0997\u09c7 \u09ac\u09cd\u09af\u09ac\u09b8\u09be \u099a\u09be\u09b2\u09c1 \u0995\u09b0\u09c7\u0964",
        "\u09b8\u09cb\u09a8\u09be\u09b0 \u09ac\u09be\u0982\u09b2\u09be \u0986\u09ae\u09bf \u09a4\u09cb\u09ae\u09be\u09af\u09bc \u09ad\u09be\u09b2\u09ac\u09be\u09b8\u09bf\u0964"
    ],
    "eval_split_max_size": null,
    "eval_split_size": 0.01,
    "use_speaker_weighted_sampler": false,
    "speaker_weighted_sampler_alpha": 1.0,
    "use_language_weighted_sampler": false,
    "language_weighted_sampler_alpha": 1.0,
    "use_length_weighted_sampler": false,
    "length_weighted_sampler_alpha": 1.0,
    "model_args": {
        "num_chars": 87,
        "out_channels": 513,
        "spec_segment_size": 32,
        "hidden_channels": 192,
        "hidden_channels_ffn_text_encoder": 768,
        "num_heads_text_encoder": 2,
        "num_layers_text_encoder": 6,
        "kernel_size_text_encoder": 3,
        "dropout_p_text_encoder": 0.1,
        "dropout_p_duration_predictor": 0.5,
        "kernel_size_posterior_encoder": 5,
        "dilation_rate_posterior_encoder": 1,
        "num_layers_posterior_encoder": 16,
        "kernel_size_flow": 5,
        "dilation_rate_flow": 1,
        "num_layers_flow": 4,
        "resblock_type_decoder": "1",
        "resblock_kernel_sizes_decoder": [
            3,
            7,
            11
        ],
        "resblock_dilation_sizes_decoder": [
            [
                1,
                3,
                5
            ],
            [
                1,
                3,
                5
            ],
            [
                1,
                3,
                5
            ]
        ],
        "upsample_rates_decoder": [
            8,
            8,
            2,
            2
        ],
        "upsample_initial_channel_decoder": 512,
        "upsample_kernel_sizes_decoder": [
            16,
            16,
            4,
            4
        ],
        "periods_multi_period_discriminator": [
            2,
            3,
            5,
            7,
            11
        ],
        "use_sdp": true,
        "noise_scale": 1.0,
        "inference_noise_scale": 0.667,
        "length_scale": 1,
        "noise_scale_dp": 1.0,
        "inference_noise_scale_dp": 1.0,
        "max_inference_len": null,
        "init_discriminator": true,
        "use_spectral_norm_disriminator": false,
        "use_speaker_embedding": false,
        "num_speakers": 0,
        "speakers_file": null,
        "d_vector_file": null,
        "speaker_embedding_channels": 256,
        "use_d_vector_file": false,
        "d_vector_dim": 0,
        "detach_dp_input": true,
        "use_language_embedding": false,
        "embedded_language_dim": 4,
        "num_languages": 0,
        "language_ids_file": null,
        "use_speaker_encoder_as_loss": false,
        "speaker_encoder_config_path": "",
        "speaker_encoder_model_path": "",
        "condition_dp_on_speaker": true,
        "freeze_encoder": false,
        "freeze_DP": false,
        "freeze_PE": false,
        "freeze_flow_decoder": false,
        "freeze_waveform_decoder": false,
        "encoder_sample_rate": null,
        "interpolate_z": true,
        "reinit_DP": false,
        "reinit_text_encoder": false
    },
    "lr_gen": 0.0002,
    "lr_disc": 0.0002,
    "lr_scheduler_gen": "ExponentialLR",
    "lr_scheduler_gen_params": {
        "gamma": 0.999875,
        "last_epoch": -1
    },
    "lr_scheduler_disc": "ExponentialLR",
    "lr_scheduler_disc_params": {
        "gamma": 0.999875,
        "last_epoch": -1
    },
    "kl_loss_alpha": 1.0,
    "disc_loss_alpha": 1.0,
    "gen_loss_alpha": 1.0,
    "feat_loss_alpha": 1.0,
    "mel_loss_alpha": 45.0,
    "dur_loss_alpha": 1.0,
    "speaker_encoder_loss_alpha": 1.0,
    "return_wav": true,
    "use_weighted_sampler": false,
    "weighted_sampler_attrs": null,
    "weighted_sampler_multipliers": null,
    "r": 1,
    "num_speakers": 0,
    "use_speaker_embedding": false,
    "speakers_file": null,
    "speaker_embedding_channels": 256,
    "language_ids_file": null,
    "use_language_embedding": false,
    "use_d_vector_file": false,
    "d_vector_file": null,
    "d_vector_dim": 0,
    "github_branch": "inside_docker"
}
```
i am attaching below my training logs

[trainer_0_log.txt.zip](https://github.com/coqui-ai/TTS/files/9866711/trainer_0_log.txt.zip)

is best_model.pth always saving best model or it is the latest checkpoint_xxx.pth that saves the best model?
i am confused when to stop training,found nothing about it in docs so i am suspecting there exist bug during saving the best model.

### To Reproduce

train vits by following this tutorial https://github.com/coqui-ai/TTS/blob/dev/notebooks/Tutorial_2_train_your_first_TTS_model.ipynb

### Expected behavior

best_model.pth should save always the best model in terms of speech quality

### Logs

```shell
i have attached above my training logs already

[trainer_0_log.txt.zip](https://github.com/coqui-ai/TTS/files/9866711/trainer_0_log.txt.zip)
```

### Environment

```shell
TTS 0.8.0
gpu : A6000
pytorch : 1.10
```

**Comment by erogol** [comment]
best_model means the best loss. Currently, we don't have/know a way to select the best_model based on voice quality which is a very subjective measure. I close this as it is not a bug.

**Comment by padmalcom** [comment]
I can definitely confirm this and it cost me some time to figure out why my model performans so badly. When the best loss does not return the best result I'd suggest not to write that file out since it does not represent what engineers expect.
